core/network/src/sync.rs

```diff
@@ -514,7 +563,8 @@ impl<B: BlockT> ChainSync<B> {
 				trace!(target: "sync", "Reversing incoming block list");
 				blocks.reverse();
 			}
-			match peer.state {
+			let peer_state = peer.state.clone();
+			match peer_state {
```

Comment:
I guess it has to be cloned because we can't borrow and then modify `peer.state`? Maybe instead this match could return the next state then to prevent that?

Reply:
I'll take a look later, I think it will be good to refactor the whole function/module.